test/test_utils.py

```diff
@@ -688,6 +688,20 @@ def test_hub_dir(self):
             torch.hub.set_dir(dirname)
             self.assertEqual(torch.hub.get_dir(), dirname)
 
+    @retry(URLError, tries=3, skip_after_retries=True)
+    def test_hub_parse_repo_info(self):
+        # If the branch is specified we just parse the input and return
+        self.assertEqual(torch.hub._parse_repo_info('a/b:c'),
+                         ('a', 'b', 'c'))
+        # For torchvision, the default branch is main
+        self.assertEqual(torch.hub._parse_repo_info('pytorch/vision'),
+                         ('pytorch', 'vision', 'main'))
+        # For linux, the default branch is still master and main doesn't exist
+        # If they ever switch to main we'll have to find another repo that uses
+        # master
+        self.assertEqual(torch.hub._parse_repo_info('torvalds/linux'),
+                         ('torvalds', 'linux', 'master'))
```

Comment:
Hum, can we mock `urlopen` so that we don't have to rely on 3rd-party repos not changing the branch to `main`?

Reply:
I'm not sure what's best here:

- if we rely on an external repo we're taking the risk of that repo switching to main eventually and we'll have to update the tes
- if we rely on a mock, we will be able to test our code's logic, but we won't be able to make sure that this logic is actually the right one. Namely if GitHub ever decides that `f"https://github.com/{repo_owner}/{repo_name}/tree/main/` doesn't return a 404 anymore, we will never catch it in the tests and torchhub will be broken without us noticing.

Instead of relying on the linux repo, maybe we could just rely on https://github.com/ailzhang/torchhub_example instead? This repo is already used in the torchhub tests in this file actually